src/jshint.js

```diff
@@ -3744,7 +3744,7 @@ var JSHINT = (function() {
     // When the if is within a for-in loop and the condition has a negative form,
     // check if the body contains nothing but a continue statement
     if (forinifcheck && forinifcheck.type === "(negative)") {
-      if (s && s.length === 1 && s[0].type === "(identifier)" && s[0].value === "continue") {
+      if (s && s[0] && s[0].type === "(identifier)" && s[0].value === "continue") {
```

Comment:
I'm all for defensive programming, but the real problem here is that `statement` is returning `undefined` for `switch` statements. Would you mind address that, too?

This also means you can simplify the test case--any `switch` statement will do.